core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -452,37 +490,114 @@ public void unmountException() throws Exception {
   }
 
   /**
-   * Ensures Warnings are logged or exceptions are thrown when an {@link AlluxioURI} with a
-   * scheme://host is passed.
+   * Ensures warnings are logged and exception is thrown when an {@link AlluxioURI} with an invalid
+   * authority is passed.
    */
   @Test
-  public void UriCheckTest() throws Exception {
+  public void uriCheckBadAuthority() throws Exception {
     Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
     Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    TestAppender appender = new TestAppender();
+    Logger.getRootLogger().addAppender(appender);
+
     AlluxioURI uri = new AlluxioURI("alluxio://localhost:1234/root");
-    boolean exceptionRaised = true;
     try {
-      BaseFileSystem.UriCheck(uri);
-      exceptionRaised = false;
+      mFileSystem.createDirectory(uri);
       fail("Should have failed on bad host and port");
     } catch (IllegalArgumentException e) {
-      assertTrue(exceptionRaised);
+      assertTrue(e.getMessage().contains(
+          "The URI host and port do not match the configured value"));
     }
 
-    uri = new AlluxioURI("alluxio://localhost:19998/root");
-    try {
-      BaseFileSystem.UriCheck(uri);
-    } catch (IllegalArgumentException e) {
-      fail("Should not have thrown an exception with a valid scheme://host:port combination.");
+    for (LoggingEvent e : appender.mEvents) {
+      if (e.getLevel().equals("WARN")) {
+        assertTrue(e.getRenderedMessage().contains("The URI schemes")
```

Comment:
could be helpful to define one helper method which reports whether any logged warning contains "The URI schemes" and another which reports whether any logged warning contains "The URI authority".

Reply:
So even if we had a method that reported whether or not the log contains the proper messages, the actual assertion would still vary by test? Having a method to report whether the proper message existed (or not) seems redundant if we want the test to fail and also report the error message it failed on because we'd need to go back through and perform the proper assertion anyways? Maybe I'm misunderstanding your suggestion

Reply:
Right, some tests want to say `I expect to see "The URI schemes"`, so they can
```java
  assertTrue(wasLogged("The URI schemes"));
```
Other tests might want to check that both messages were logged, so they could say
```java
  assertTrue(wasLogged("The URI schemes"));
  assertTrue(wasLogged("The URI authority"));
```
Other tests might want to check that the messages were not logged
```java
  assertFalse(wasLogged("The URI schemes"));
  assertFalse(wasLogged("The URI authority"));
```

The helper method saves the trouble of having to iterate over all the logging events each time, and makes the intent of the test easier to follow